src/main/java/org/dependencytrack/resources/v1/ProjectResource.java

```diff
@@ -275,6 +280,99 @@ public Response updateProject(Project jsonProject) {
+                if (modified && qm.getProject(project.getName(), project.getVersion()) != null) {
+                    return Response.status(Response.Status.CONFLICT).entity("A project with the specified name and version already exists.").build();
+                }
+                modified |= copyIfDifferent(jsonProject, project, Project::getAuthor, Project::setAuthor);
+                modified |= copyIfDifferent(jsonProject, project, Project::getPublisher, Project::setPublisher);
+                modified |= copyIfDifferent(jsonProject, project, Project::getGroup, Project::setGroup);
+                modified |= copyIfDifferent(jsonProject, project, Project::getDescription, Project::setDescription);
+                modified |= copyIfDifferent(jsonProject, project, Project::getClassifier, Project::setClassifier);
+                modified |= copyIfDifferent(jsonProject, project, Project::getCpe, Project::setCpe);
+                modified |= copyIfDifferent(jsonProject, project, Project::getPurl, Project::setPurl);
+                modified |= copyIfDifferent(jsonProject, project, Project::getSwidTagId, Project::setSwidTagId);
+                modified |= copyIfDifferent(jsonProject, project, Project::isActive, Project::setActive);
+                if (jsonProject.getTags() != null && (!Collections.isEmpty(jsonProject.getTags()) || !Collections.isEmpty(project.getTags()))) {
+                    modified = true;
+                    project.setTags(jsonProject.getTags());
+                }
+                if (modified) {
+                    project = qm.updateProject(project, true);
+                    return Response.ok(project).build();
+                } else {
+                    return Response.notModified().build();
+                }
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).entity("The UUID of the project could not be found.").build();
+            }
+        }
+    }
+
+    /**
+     * updates the given target object using the supplied setter method with the
+     * new value from the source object using the supplied getter method. But
+     * only if the new value is not {@code null} and it is not
+     * {@link Object#equals(java.lang.Object) equal to} the old value.
+     *
+     * @param <T> the type of the old and new value
+     * @param source the source object that contains the new value
+     * @param target the target object that should be updated
+     * @param getter the method to retrieve the new value from {@code source}
+     * and the old value from {@code target}
+     * @param setter the method to set the new value on {@code target}
+     * @return {@code true} if {@code target} has been changed, else
+     * {@code false}
+     */
+    private <T> boolean copyIfDifferent(final Project source, final Project target, final Function<Project, T> getter, final BiConsumer<Project, T> setter) {
```

Comment:
Nitpick: It's less copying and more setting. Rename this to `setIfDifferent` maybe?